application-workloads/sap/sap-ilm-store/azuredeploy.json

```diff
@@ -0,0 +1,181 @@
+        "Standard_RAGZRS"
+      ]
+    },
+    "storageSKUTier": {
+      "type": "string",
+      "defaultValue": "Standard",
+      "metadata": {
+        "description": "Performance tiers for block blob storage"
+      },
+      "allowedValues": [
+        "Standard",
+        "Premium"
+      ]
+    },
+    "storageKind": {
+      "type": "string",
+      "defaultValue": "StorageV2",
+      "metadata": {
+        "description": "Type of storage account"
+      },
+      "allowedValues": [
+        "Storage",
+        "StorageV2",
+        "BlobStorage",
+        "FileStorage",
+        "BlockBlobStorage"
+      ]
+    },
+    "accessTier": {
+      "type": "string",
+      "defaultValue": "Hot",
+      "metadata": {
+        "description": "Access tiers for Azure Blob Storage"
+      },
+      "allowedValues": [
+        "Hot",
+        "Cool"
+      ]
+    }
+  },
+  "variables": {
+    "roleDefinitionId": "[guid(subscription().id, string(parameters('customRoleName')))]"
+  },
+  "resources": [
+    {
+      "type": "Microsoft.Storage/storageAccounts",
+      "apiVersion": "2021-02-01",
+      "name": "[parameters('storageAccountName')]",
+      "location": "[parameters('location')]",
+      "sku": {
+        "name": "[parameters('storageSKUName')]",
+        "tier": "[parameters('storageSKUTier')]"
+      },
+      "kind": "[parameters('storageKind')]",
+      "properties": {
+        "accessTier": "[parameters('accessTier')]",
+        "allowSharedKeyAccess": false,
+        "supportsHttpsTrafficOnly": true,
+        "allowBlobPublicAccess": false,
+        "minimumTlsVersion": "TLS1_2"
+      }
+    },
+    {
+      "condition": "[equals(parameters('newOrExistingRole'),'new')]",
+      "type": "Microsoft.Authorization/roleDefinitions",
+      "apiVersion": "2018-07-01",
+      "name": "[variables('roleDefinitionId')]",
+      "properties": {
+        "roleName": "[parameters('customRoleName')]",
+        "description": "Role definition for ILM specific AD Application to access a Storage Account",
+        "type": "customRole",
+        "isCustom": true,
+        "permissions": [
+          {
+            "actions": [
+              "Microsoft.Storage/storageAccounts/blobServices/containers/write",
+              "Microsoft.Storage/storageAccounts/blobServices/containers/delete"
+            ],
+            "dataActions": [
+              "Microsoft.Storage/storageAccounts/blobServices/containers/blobs/read",
+              "Microsoft.Storage/storageAccounts/blobServices/containers/blobs/write",
+              "Microsoft.Storage/storageAccounts/blobServices/containers/blobs/delete"
+            ]
+          }
+        ],
+        "assignableScopes": [
+          "[concat('/subscriptions/', subscription().subscriptionId, '/resourcegroups/', resourceGroup().name)]"
+        ]
+      }
+    },
+    {
+      "type": "Microsoft.Authorization/roleAssignments",
+      "apiVersion": "2018-07-01",
+      "name": "[guid(resourceGroup().id)]",

```

Comment:
this needs to be idempotent and use principal, role and scope in the seed.

guid(parameters('azureADAppPrincipalID'), if(equals(parameters('newOrExistingRole'), 'new'), resourceId('Microsoft.Authorization/roleDefinitions/', variables('roleDefinitionId')), resourceId('Microsoft.Authorization/roleDefinitions/', parameters('existingRoleID'))), resourceGroup().id)

Reply:
Made the necessary changes